aws/application_load_balancer/main.tf

```diff
@@ -29,34 +29,42 @@ module "load_balancer" {
 resource "aws_security_group" "security_group_on_load_balancer" {
   name   = "${local.name_prefix}-alb"
   vpc_id = "${var.vpc_id}"
+}
 
-  egress {
-    from_port       = "${local.http_port_for_instances}"
-    protocol        = "tcp"
-    security_groups = ["${var.security_group_for_instances}"]
-    to_port         = "${local.http_port_for_instances}"
-  }
+resource "aws_security_group_rule" "http_egress_on_load_balancer" {
```

Comment:
Changing these from inline blocks to `aws_security_group_rule` will make it impossible to safely update users of this module because of Terraform’s lack of a migration plan for these security groups 😞 Every time I’ve looked at doing this, there hasn’t been a way…

Reply:
I assume you have tried this already, but could we target the `aws_security_group` change first and then add the aws_security_group_rule on a second pass?

Reply:
I think I haven’t done that before, because it would temporarily leave the security group without any rules, so access would be blocked until the second pass (which would add the rules back), no?